specutils/spectra/xrayspectrum1d.py

```diff
@@ -0,0 +1,542 @@
+            * `n_chan` stores the number of channels in each channel
+              set
+
+        As a result, for a given energy bin i, we need to look up the
+        number of channel sets in `n_grp` for that energy bin. We
+        then need to loop over the number of channel sets. For each
+        channel set, we look up the first channel into which flux
+        will be distributed as well as the number of channels in the
+        group. We then need to also loop over the these channels and
+        actually use the corresponding elements in the redistribution
+        matrix to redistribute the photon flux into channels.
+
+        All of this is basically a big bookkeeping exercise in making
+        sure to get the indices right.
+
+        Parameters
+        ----------
+        spec : numpy.ndarray
+            The (model) spectrum to be folded
+
+        Returns
+        -------
+        counts : numpy.ndarray
+            The (model) spectrum after folding, in
+            counts/s/channel
+
+        """
+        # get the number of channels in the data
+        nchannels = spec.shape[0]
+
+        # an empty array for the output counts
+        counts = np.zeros(nchannels)
+
+        # index for n_chan and f_chan incrementation
+        k = 0
+
+        # index for the response matrix incrementation
+        resp_idx = 0
+
+        # loop over all channels
+        for i in range(nchannels):
+
+            # this is the current bin in the flux spectrum to
+            # be folded
+            source_bin_i = spec[i]
+
+            # get the current number of groups
+            current_num_groups = self.n_grp[i]
+
+            # loop over the current number of groups
+            for j in range(current_num_groups):
+
+                current_num_chans = int(self.n_chan[k])
+
+                if current_num_chans == 0:
+                    k += 1
+                    resp_idx += current_num_chans
+                    continue
+
+
+                else:
+                    # get the right index for the start of the counts array
+                    # to put the data into
+                    counts_idx = int(self.f_chan[k] - self.offset)
+                    # this is the current number of channels to use
+
+                    k += 1
+                    # add the flux to the subarray of the counts array that starts with
+                    # counts_idx and runs over current_num_chans channels
+                    counts[counts_idx:counts_idx +
+                                      current_num_chans] += self.matrix[resp_idx:resp_idx +
+                                                                                 current_num_chans] * \
+                                                                np.float(source_bin_i)
+                    # iterate the response index for next round
+                    resp_idx += current_num_chans
+
+
+        return counts[:self.detchans]
+
+
+class ARF(object):
+
```

Comment:
Please spell out the full name here for the class

Reply:
Changed to AreaResponse